nowcasting_api/national.py

```diff
@@ -162,6 +162,34 @@ def get_national_forecast(
             format_plevels(national_forecast_value)
 
             national_forecast_values.append(national_forecast_value)
+
+    # Adjust values for solar eclipse 2025-03-28
+    eclipse_adjustments = {
+        "2025-03-29 10:30:00+00:00": 200,
+        "2025-03-29 11:00:00+00:00": 1000,
+        "2025-03-29 11:30:00+00:00": 1500,
+        "2025-03-29 12:00:00+00:00": 400,
+    }
+
+    # Reduce national forecast value by eclipse adjustment for matching datetimes
+    for forecast_value in national_forecast_values:
+        target_time = forecast_value.target_time
+        target_dt = pd.to_datetime(target_time)
+
+        for eclipse_time, reduction in eclipse_adjustments.items():
+            if target_dt == pd.to_datetime(eclipse_time):
+                forecast_value.expected_power_generation_megawatts -= reduction
+
+    if get_plevels:
+        for forecast_value in national_forecast_values:
+            target_time = forecast_value.target_time
+            target_dt = pd.to_datetime(target_time)
+
+            for eclipse_time, reduction in eclipse_adjustments.items():
+                if target_dt == pd.to_datetime(eclipse_time):
+                    forecast_value.plevels["plevel_10"] -= reduction
```

Comment:
% also possibly makes these slightly more representative as well, though that is murkier with probabilistic things...